### PR TITLE
Docstrings for RecordingExtractorDataChunkIterator

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

Enhancing docstrings for the `RecordingExtractorDataChunkIterator` class.

## How to test the behavior?

No change to behavior.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

**Comment by bendichter** [comment]
@CodyCBakerPhD thanks for drafting this. This is too wordy and not precise enough. You need to indicate that shape and size parameters cannot be specified at the same time.

**Comment by CodyCBakerPhD** [comment]
@bendichter True that; simplified text a bit and mentioned the constraint on shape/size.

File: nwb_conversion_tools/utils/recordingextractordatachunkiterator.py [hunk 1]
```diff
@@ -17,6 +17,34 @@ def __init__(
...
+        buffer_shape : tuple, optional
+            Manual specification of buffer shape to return on each iteration.
+            Must be a multiple of chunk_shape along each axis.
+            Cannot be set if `buffer_gb` is also specified.
+            The default is None.
+        chunk_mb : float, optional
+            The upper bound on size in megabytes (MB) of the internal chunk for the HDF5 dataset.
+            The chunk_shape will be set implicitly by this argument.
+            Cannot be set if `chunk_shape` is also specified.
+            The default is 1MB, which is highly recommended by the HDF5 group. For more details, see
```
Comment (bendichter) [inline nwb_conversion_tools/utils/recordingextractordatachunkiterator.py:41]:
```suggestion
            The default is 1MB, as recommended by the HDF5 group. For more details, see
```

**Comment by bendichter** [comment]
Thanks this is a big improvement! Looks good.

**Review by bendichter — APPROVED** [review]
